### ISSUE TITLE
Row Key - Nested Data or Auto-Generate Support

### ISSUE THREAD

**Issue by alyssaBiasi** [reactabular/reactabular#193]
I have found a need for `rowKey` to either be:

- Able to access nested data (e.g. `dependency.toThing.name` from my example below)
- Auto-generated based on the data index number

Sorry if this already exists. I couldn't find mention of it in the docs (I even went source code diving to figure it out)

### Context
I'm trying to render a table of dependencies between objects.

For example, if I have a bunch of  `Thing`s:

```
class Thing {
    constructor(name) {
         this.name = name
    }
}
```

`Thing`s have dependencies between each other and some data about what the relationship is (e.g. "Thing1" depends on "Thing2" for some data A and depends on "Thing3" for some data B):

```
class Dependency {
    constructor(fromThing, toThing, data) {
         this.fromThing = fromThing
         this.toThing = toThing
         this.data = data
    }
}
```

Despite `dependency.toThing.name` being unique for a set of  `Dependency `s, I can't access it through the `rowKey` prop. In order to render a table of `Dependency `s for "Thing1", I have to generate a meaningless attribute on the `Dependency` definition in order to use it as the table `rowKey`.

**Comment by bebraw** [comment]
Hi,

Thanks for the report. I think I'll expand the interface so that you can pass a function to it that can resolve the key from data. That should allow you to implement a more complicated lookup.

I have implemented something specific for [resolving data](http://reactabular.js.org/#/resolving-data) in particular. Especially `resolve.byFunction` could fit your case. Another option would be to resolve the id through this mechanism and then use the current API. I think I'll make the interface more flexible regardless.

**Comment by bebraw** [comment]
Included in `2.0.5`. I hope it works for you now.

**Comment by alyssaBiasi** [comment]
Legend. Thanks 😄
